### ISSUE TITLE
Weather entity unavailable

### ISSUE THREAD

**Issue by filmgarage** [golles/ha-knmi#51]
### Did you read the instructions?

- [X] I have read the [README](../blob/main/README.md), including the [known-limitations](../blob/main/README.md#known-limitations) section.

### The problem

The entity weather.my_location is unavailable since the last update (1.3.2) but could be missing on an earlier version. I noticed it today. Reinstalling the integration does not solve this and adds the unavailable entity.

The following error comes up with the entity:

This entity is no longer being provided by the knmi integration. If the entity is no longer in use, delete it in settings.

### What version of this integration has the issue?

1.3.2

### What version of Home Assistant Core has the issue?

2022.11.3

### Home Assistant log

```shell
Logger: homeassistant.components.weather
Source: custom_components/knmi/__init__.py:111 
Integration: Weather (documentation, issues) 
First occurred: 23:00:12 (2 occurrences) 
Last logged: 23:00:12

Error adding entities for domain weather with platform knmi
Error while setting up knmi platform for weather
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 692, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 789, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 585, in _async_write_ha_state
    attr.update(self.state_attributes or {})
  File "/usr/src/homeassistant/homeassistant/components/weather/__init__.py", line 710, in state_attributes
    if self.forecast is not None:
  File "/config/custom_components/knmi/weather.py", line 150, in forecast
    wind_bearing = self.coordinator.get_value(f"d{i}windrgr", float)
  File "/config/custom_components/knmi/__init__.py", line 111, in get_value
    return convert_to(self.data.get(key, None))
ValueError: could not convert string to float: ''
```

### Diagnostics

```json
Not available
```

**Comment by marcschut002** [comment]
I have the same problem

**Comment by dutch-erik** [comment]
Same error indeed.

**Comment by RichardvanGestel** [comment]
me too

**Comment by golles** [comment]
There seems to have been an issue with the API between 2022-11-17 17:06 and 2022-11-18 11:06. 

The component should've automatically recovered from this issue. I'll make a fix for this, handling these errors properly.

**Comment by diedvdyk** [comment]
I got this one :

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 813, in _update_entity_states
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 532, in async_update_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 583, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 551, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/weather/__init__.py", line 832, in state
    return self.condition
  File "/config/custom_components/knmi/weather.py", line 102, in condition
    return CONDITIONS_MAP[self.coordinator.get_value("image")]
KeyError: 'nachtbewolkt'

**Comment by golles** [comment]
> I got this one :
> 
> Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 813, in _update_entity_states await asyncio.gather(*tasks) File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 532, in async_update_ha_state self._async_write_ha_state() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 583, in _async_write_ha_state state = self._stringify_state(available) File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 551, in _stringify_state if (state := self.state) is None: File "/usr/src/homeassistant/homeassistant/components/weather/**init**.py", line 832, in state return self.condition File "/config/custom_components/knmi/weather.py", line 102, in condition return CONDITIONS_MAP[self.coordinator.get_value("image")] KeyError: 'nachtbewolkt'

This is one different, this will be fixed in #52

**Comment by dutch-erik** [comment]
`Logger: homeassistant.components.weather
Source: custom_components/knmi/weather.py:103 
Integration: Weather (documentation, issues) 
First occurred: 17:33:37 (6 occurrences) 
Last logged: 19:46:10

Error adding entities for domain weather with platform knmi
Error while setting up knmi platform for weather
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 692, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 789, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 583, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 551, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/weather/__init__.py", line 832, in state
    return self.condition
  File "/config/custom_components/knmi/weather.py", line 103, in condition
    return CONDITIONS_MAP[self.coordinator.get_value("image")]
KeyError: 'wolkennacht'
`
Is there again an issue with the api?

**Comment by golles** [comment]
> ```
> Logger: homeassistant.components.weather
> Source: custom_components/knmi/weather.py:103 
> Integration: Weather (documentation, issues) 
> First occurred: 17:33:37 (6 occurrences) 
> Last logged: 19:46:10
> 
> Error adding entities for domain weather with platform knmi
> Error while setting up knmi platform for weather
> Traceback (most recent call last):
>   File "/usr/src/homeassistante/homeassistant/helpers/entity_platform.py", line 428, in async_add_entities
>     await asyncio.gather(*tasks)
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 692, in _async_add_entity
>     await entity.add_to_platform_finish()
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 789, in add_to_platform_finish
>     self.async_write_ha_state()
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 545, in async_write_ha_state
>     self._async_write_ha_state()
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 583, in _async_write_ha_state
>     state = self._stringify_state(available)
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 551, in _stringify_state
>     if (state := self.state) is None:
>   File "/usr/src/homeassistant/homeassistant/components/weather/__init__.py", line 832, in state
>     return self.condition
>   File "/config/custom_components/knmi/weather.py", line 103, in condition
>     return CONDITIONS_MAP[self.coordinator.get_value("image")]
> KeyError: 'wolkennacht'
> ```
> 
> Is there again an issue with the api?

Please create a new issue for that with the full `home-assistant.log` and diagnostics file

**Comment by dutch-erik** [comment]
Just noticed another user @haaino opened new issue for it. 😉
Issue was ‘solved’ for me last night after restart, while other restarts during the day didn’t.

**Comment by golles** [comment]
Both issues should be fixed in 1.3.4
